From 6b85029c314f5a951db7bd4f919932b7c9135c11 Mon Sep 17 00:00:00 2001
From: Ranjani Vaidyanathan-RA5478 <Ranjani.Vaidyanathan@freescale.com>
Date: Mon, 26 Jan 2009 11:22:04 -0600
Subject: [PATCH] ENGR00106273-2: MX51: Change the default CPU freq to be 800MHz

Added 800MHz working point for CPU frequency.
Fixed the bug that was causing issues with changing the CPU frequency between 200MHz
and 800MHz.

Signed-off-by: Ranjani Vaidyanathan-RA5478 <Ranjani.Vaidyanathan@freescale.com>
---
 arch/arm/mach-mx51/clock.c       |    3 ++-
 arch/arm/mach-mx51/mx51_3stack.c |   12 ++++++------
 2 files changed, 8 insertions(+), 7 deletions(-)

diff --git a/arch/arm/mach-mx51/clock.c b/arch/arm/mach-mx51/clock.c
index d94c971..2c872fe 100644
--- a/arch/arm/mach-mx51/clock.c
+++ b/arch/arm/mach-mx51/clock.c
@@ -44,10 +44,11 @@ static struct clk ipu_clk[];
 static struct clk axi_a_clk;
 static struct clk axi_b_clk;
 static struct clk ddr_hf_clk;
-static int cpu_wp_nr;
 static int cpu_curr_wp;
 static struct cpu_wp *cpu_wp_tbl;
 
+int cpu_wp_nr;
+
 extern int mxc_jtag_enabled;
 
 static int cpu_clk_set_wp(int wp);
diff --git a/arch/arm/mach-mx51/mx51_3stack.c b/arch/arm/mach-mx51/mx51_3stack.c
index 8ab3a81..e486ab8 100644
--- a/arch/arm/mach-mx51/mx51_3stack.c
+++ b/arch/arm/mach-mx51/mx51_3stack.c
@@ -67,15 +67,15 @@ extern int mxc_clocks_init(void);
 extern void __init early_console_setup(char *);
 extern int mxc_init_devices(void);
 
-/* working point(wp): 0 - 665MHz; 1 - 200MHz; */
+/* working point(wp): 0 - 800MHz; 1 - 200MHz; */
 static struct cpu_wp cpu_wp_auto[] = {
 	{
-	 .pll_rate = 665000000,
-	 .cpu_rate = 665000000,
+	 .pll_rate = 800000000,
+	 .cpu_rate = 800000000,
 	 .pdf = 0,
-	 .mfi = 6,
-	 .mfd = 95,
-	 .mfn = 89,
+	 .mfi = 8,
+	 .mfd = 2,
+	 .mfn = 1,
 	 .cpu_voltage = 1050000,},
 	{
 	 .pll_rate = 200000000,
-- 
1.5.4.4

